frontend/public/components/factory/table.tsx

```diff
@@ -276,15 +276,52 @@ export type TableRowProps = {
   className?: string;
 };
 
+const BREAKPOINT_SM = 576;
+const BREAKPOINT_MD = 768;
+const BREAKPOINT_LG = 992;
+const BREAKPOINT_XL = 1200;
+const BREAKPOINT_XXL = 1400;
+const MAX_COL_SM = 2;
+const MAX_COL_MD = 4;
+const MAX_COL_LG = 4;
+const MAX_COL_XL = 6;
+const MAX_COL_XXL = 8;
+
+const isColumnVisible = (columnID: string, columns: Set<string> = new Set()) => {
+  const showNamespace =
+    columnID !== 'namespace' || UIActions.getActiveNamespace() === ALL_NAMESPACES_KEY;
+  if (_.isEmpty(columns) && showNamespace) {
+    return true;
+  }
+  if (!columns.has(columnID)) {
+    return false;
+  }
+  const widthInPixels = window.innerWidth;
+  if (widthInPixels < BREAKPOINT_SM) {
+    return [...columns].indexOf(columnID) < MAX_COL_SM;
```

Comment:
I think these names are confusing because if we're `< 576`, we're not at small. We're at extra small. But we're checking `MAX_COL_SM`. Should it be `MAX_COL_XS`, etc?

Reply:
@spadgett good point, its one size lower than named. fixed this.